### PR TITLE
New and Optimised Splash Screen

### LINKED ISSUE #114: Make an Awesome Splash Screen
**Actual Behaviour**
Currently, there is no Splash Screen for the app.

**Expected Behaviour**
The Splash screen will make the app more attractive and meaningful from the launch of the app.

**Steps to reproduce it**
This feature should be implemented when the user launch the app. The launch time will be 3000 milliseconds.

**Sample Screenshot**
Splash Screen

**Would you like to work on the issue?**
Yes, I would like to work on this issue 👍

### PR DESCRIPTION
**Pull request by vishal-sehgal**
Fixes #114 

**Changes**: 
I have improved and optimize the splash screen launch which is now more meaningful with the app's name.

**Gif for the changes**: 
SplashScreen-Neurolab Demo

**Checklist**:
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications are done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[#114 feature splash screen.apk.zip](https://github.com/fossasia/neurolab-android/files/3022646/114.feature.splash.screen.apk.zip)

### CODE REVIEW

**Comment by vishal-sehgal** [comment]
@CloudyPadmal @iamareebjamal Can you please look into this PR?

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -8,24 +9,28 @@
...
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
         <activity
             android:name=".ProgramModeActivity"
-            android:parentActivityName=".MainActivity"/>
+            android:parentActivityName=".MainActivity" />
```
Comment (jddeep) [inline app/src/main/AndroidManifest.xml:24]: Why this change? Please remove.

**Review by jddeep — CHANGES_REQUESTED** [review]

**Review by jddeep — COMMENTED** [review]
We already had a splash screen in the app. As of now please focus on functionalities over UI. The UX/UI definitely needs work but only after implementing core features and functionalities.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (vishal-sehgal) [reply app/src/main/AndroidManifest.xml:24]: I just reformatted the code by pressing CTRL+ALT+L nothing else.

**Comment by vishal-sehgal** [comment]
@jddeep Actually I knew that the app already had a Splash Screen but I thought to improve it along with other functionalities because it will really give a great experience while running the app but as you're saying to focus on functionalities first so I will focus on them for now 👍

File: app/src/main/java/io/neurolab/SplashActivity.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_splash);
+
+        new Handler().postDelayed(() -> {
+            startActivity(new Intent(SplashActivity.this, MainActivity.class));
+            finish();
+        }, SPLASH_SCREEN_DELAYED_TIME);
```
Comment (iamareebjamal) [inline app/src/main/java/io/neurolab/SplashActivity.java:21]: Horrible idea. Why should I wait 3 seconds more to open the app?

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (suresh-gandhi) [reply app/src/main/AndroidManifest.xml:24]: @jddeep Reformatting is necessary and don't worry this won't cause merge conflicts.

**Comment by suresh-gandhi** [comment]
@CoderVishalSehgal Its evident that you have put a lot of time and effort into this PR and I really appreciate it. However it would be great if you take the following changes into account. I see that you have implemented SplashActivity for the splash screen. This approach was actually used before. Nowadays, Google's material design specification specifies a new optimized pattern for creating splash screens by utilizing the time lag between when the app is launched but isn't in the memory yet. You can study more about this pattern [here](https://android.jlelse.eu/right-way-to-create-splash-screen-on-android-e7f1709ba154). 

As you can see @RohitJain1103 already used this pattern while making a splash screen for our application. See [this](https://github.com/fossasia/neurolab-android/pull/53) PR. What you can do is improve the look and feel of this splash screen(For instance it would be really great if you could gif's in our splash screen like fading in/out for a brief time instead of a single static picture). For that you may be requiring support of external dependencies like Glide perhaps.

File: app/src/main/java/io/neurolab/SplashActivity.java (hunk 2 above)
Comment (vishal-sehgal) [reply app/src/main/java/io/neurolab/SplashActivity.java:21]: Sorry I will change it to 1 second.

**Comment by vishal-sehgal** [comment]
@suresh-gandhi Thank you very much! I already know about that method actually that was just the heck just to impose the white screen(we can't even customised that rather than a static drawable)  and it was not an official Google's material design implementation and with my own implementation, I fixed that.
What Splash Screen actually is? 
It is not just a static screen(in this case, I agree I made it but In future it'll be really helpful, believe me) it is used to perform some background task for ex.- Performing any network requests or fetching some data in the meantime of loading splash screen or whatever not?! 
You can even check out my live apps [here](https://play.google.com/store/apps/dev?id=5510012331358915197) where I am using the same pattern with no chance of white screen and super fast and it doesn't mean that it's an old pattern :( 
I already have some amazing animations in my mind for SplashScreen for our project which I'll implement in the future. Believe me, the separate splash screen will be really helpful that is why I made it and I thought some of you know it already. 
And Open Source Contributions is all about to learn new things every time from each other and issue is issue whether it is small or a big :)

@iamareebjamal @suresh-gandhi I am pretty sure you guys even didn't try the attached sample apk, Please give it a try and feel it within your hands and If you don't feel it is not good enough, I'll not again ask for this issue :)

**Comment by iamareebjamal** [comment]
Not going to approve a method which uses Splash Activity, it is an anti-pattern and a terrible user experience. There are 0 benefits of using another Activity and making the user wait for no reason.

**Comment by mariobehling** [comment]
Closing this due to inactivity.
